### DISCUSSION TITLE
InvalidateQueries doesn't work after mutation

### DISCUSSION THREAD

**Discussion by dgd03146** [TanStack/query#5479] [Q&A]
### Describe the bug

I made the code in Next.js which creates the post but it doesn't seem to refetch.
It works only when I refreshed the page.

I don't understand why it doesn't work. If somebody knows I'd be happy to hear you.

### Your minimal, reproducible example

https://codesandbox.io/s/invalidatequries-doesnt-work-after-mutation-yuh34q?file=/src/useAddPost.ts

### Steps to reproduce

1. I added `async await` in `invalidateQueries` but it didn't work out.
2. I used `queryClient` with `useQueryClient` instead of using `new QueryClient` to share the same instance but it didn't work out as well

```tsx
const { mutate: onAdd } = useMutation(createPost, {
    onSuccess: async () => {
      await queryClient.invalidateQueries([QueryKeys.posts]);
      await queryClient.invalidateQueries([QueryKeys.myPostsList]);

      showToast({
        type: 'success',
        message: 'Success to create a post! '
      });
      return router.push('/posts'); // Redirect to /posts
    },
    onError: () => {
      showToast({
        type: 'error',
        message: 'Fail to create a post!'
      });
    }
  });
```
Here is my useQuery code by using useInfiniteQuery for infinite scroll.
It uses same query key `QueryKeys.posts`

 ```tsx
 const {
    data: posts,
    fetchNextPage,
    isFetchingNextPage,
    isFetching,
    isLoading
  } = useInfiniteQuery(
    [QueryKeys.posts, areaSelected, search],
    ({ pageParam = 0 }) => getPosts(pageParam, areaSelected, search),
    {
      getNextPageParam: (lastPage) =>
        !lastPage?.last ? lastPage?.nextPage : undefined,
      staleTime: 60000,
      onError(err) {
        if (err instanceof AxiosError)
          showToast({ type: 'error', message: err.message });
      }
    }
  );
```

### Expected behavior

This should be refetched when navigates to '/posts'

### How often does this bug happen?

Every time

### Platform

- OS: Window
- Version: Chrome

### Tanstack Query adapter

None

### TanStack Query version

4.0.10

**Comment by baeharam** [comment]
Your reproduction example doesn't have specific query code. I think you missed it 🤔

**Comment by dgd03146** [comment]
> Your reproduction example doesn't have specific query code. I think you missed it 🤔

I fixed it could you check again?

**Comment by baeharam** [comment]
@dgd03146 

> I fixed it could you check again?

`usePosts` and `useAddPosts` are empty.
